pkg/resourcegroup/controllers/status/status.go

```diff
@@ -133,7 +133,11 @@ func GetSourceHash(annotations map[string]string) string {
 	if len(annotations) == 0 {
 		return ""
 	}
-	sourceHash := annotations[SourceHashAnnotationKey]
+	return TruncateSourceHash(annotations[SourceHashAnnotationKey])
+}
+
+// TruncateSourceHash truncates the provided source hash after the first 7 characters.
+func TruncateSourceHash(sourceHash string) string {
```

Comment:
nit: Is there a better place for this than under the controllers package? Like maybe under pkg/resourcegroup?

Reply:
Done